code/modules/reagents/chemistry/recipes/pyrotechnics.dm

```diff
@@ -411,9 +411,9 @@ datum/chemical_reaction/flash_powder
 
 /datum/chemical_reaction/shock_explosion/on_reaction(datum/reagents/holder, created_volume)
 	var/turf/T = get_turf(holder.my_atom)
-	for(var/mob/living/carbon/C in view(6, T))
+	for(var/mob/living/carbon/C in view(max(1, round(created_volume / 3)), T))
```

Comment:
Example case for teslium explosions:
To get the same range of effect:
`total reagents needed = (5+5)*(required created_volume) = 10*(3*6) = 180 units`